File: tests/__file_snapshots__/Person.R
#` A person (alive, dead, undead, or fictional).
#`
#` @param  address Postal address for the person.
#` @param  affiliations Organizations that the person is affiliated with.
#` @param  alternateNames Alternate names (aliases) for the item.
#` @param  description A description of the item.
#` @param  emails Email addresses for the person.
#` @param  familyNames Family name. In the U.S., the last name of a person.
#` @param  funders A person or organization that supports (sponsors) something through some kind of financial contribution.
#` @param  givenNames Given name. In the U.S., the first name of a person.
#` @param  honorificPrefix An honorific prefix preceding a person's name such as Dr/Mrs/Mr.
#` @param  honorificSuffix An honorific suffix after a person's name such as MD/PhD/MSCSW.
#` @param  id The identifier for this item.
#` @param  jobTitle The job title of the person (for example, Financial Manager).
#` @param  memberOf An organization (or program membership) to which this person belongs.
#` @param  meta Metadata associated with this item.
#` @param  name The name of the item.
#` @param  telephoneNumbers Telephone numbers for the person.
#` @param  url The URL of the item.
#` @export
Person <- function (
  address,
  affiliations,
  alternateNames,
  description,
  emails,
  familyNames,
  funders,
  givenNames,
  honorificPrefix,
  honorificSuffix,
  id,
  jobTitle,
  memberOf,
  meta,
  name,
  telephoneNumbers,
  url
){
  self <- Thing(
    alternateNames=alternateNames,
    description=description,
    id=id,
    meta=meta,
    name=name,
    url=url
  )
  if(!missing(address)) setProp(self, "address", "character", address)
  if(!missing(affiliations)) setProp(self, "affiliations", Array("Organization"), affiliations)
  if(!missing(emails)) setProp(self, "emails", Array("character"), emails)
  if(!missing(familyNames)) setProp(self, "familyNames", Array("character"), familyNames)
  if(!missing(funders)) setProp(self, "funders", Array(Union("Organization", "Person")), funders)
  if(!missing(givenNames)) setProp(self, "givenNames", Array("character"), givenNames)
  if(!missing(honorificPrefix)) setProp(self, "honorificPrefix", "character", honorificPrefix)
  if(!missing(honorificSuffix)) setProp(self, "honorificSuffix", "character", honorificSuffix)
  if(!missing(jobTitle)) setProp(self, "jobTitle", "character", jobTitle)
  if(!missing(memberOf)) setProp(self, "memberOf", Array("Organization"), memberOf)
  if(!missing(telephoneNumbers)) setProp(self, "telephoneNumbers", Array("character"), telephoneNumbers)
  class(self) <- c(class(self), "Person")
  self
}

File: schema/src/bindings/__file_snapshots__/Person.py
class Person(Thing):
    """A person (alive, dead, undead, or fictional)."""

    address: Optional[Union["PostalAddress", String]] = None
    """Postal address for the person."""

    affiliations: Optional[Array["Organization"]] = None
    """Organizations that the person is affiliated with."""

    emails: Optional[Array[String]] = None
    """Email addresses for the person."""

    familyNames: Optional[Array[String]] = None
    """Family name. In the U.S., the last name of a person."""

    funders: Optional[Array[Union["Organization", "Person"]]] = None
    """A person or organization that supports (sponsors) something through
some kind of financial contribution.
"""

    givenNames: Optional[Array[String]] = None
    """Given name. In the U.S., the first name of a person."""

    honorificPrefix: Optional[String] = None
    """An honorific prefix preceding a person's name such as Dr/Mrs/Mr."""

    honorificSuffix: Optional[String] = None
    """An honorific suffix after a person's name such as MD/PhD/MSCSW."""

    jobTitle: Optional[String] = None
    """The job title of the person (for example, Financial Manager)."""

    memberOf: Optional[Array["Organization"]] = None
    """An organization (or program membership) to which this person belongs."""

    telephoneNumbers: Optional[Array[String]] = None
    """Telephone numbers for the person."""


    def __init__(
        self,
        address: Optional[Union["PostalAddress", String]] = None,
        affiliations: Optional[Array["Organization"]] = None,
        alternateNames: Optional[Array[String]] = None,
        description: Optional[Union[Array["BlockContent"], Array["InlineContent"], String]] = None,
        emails: Optional[Array[String]] = None,
        familyNames: Optional[Array[String]] = None,
        funders: Optional[Array[Union["Organization", "Person"]]] = None,
        givenNames: Optional[Array[String]] = None,
        honorificPrefix: Optional[String] = None,
        honorificSuffix: Optional[String] = None,
        id: Optional[String] = None,
        identifiers: Optional[Array[Union["PropertyValue", String]]] = None,
        images: Optional[Array[Union["ImageObject", String]]] = None,
        jobTitle: Optional[String] = None,
        memberOf: Optional[Array["Organization"]] = None,
        meta: Optional[Object] = None,
        name: Optional[String] = None,
        telephoneNumbers: Optional[Array[String]] = None,
        url: Optional[String] = None
    ) -> None:
        super().__init__(
            alternateNames=alternateNames,
            description=description,
            id=id,
            identifiers=identifiers,
            images=images,
            meta=meta,
            name=name,
            url=url
        )
        if address is not None:
            self.address = address
        if affiliations is not None:
            self.affiliations = affiliations
        if emails is not None:
            self.emails = emails
        if familyNames is not None:
            self.familyNames = familyNames
        if funders is not None:
            self.funders = funders
        if givenNames is not None:
            self.givenNames = givenNames
        if honorificPrefix is not None:
            self.honorificPrefix = honorificPrefix
        if honorificSuffix is not None:
            self.honorificSuffix = honorificSuffix
        if jobTitle is not None:
            self.jobTitle = jobTitle
        if memberOf is not None:
            self.memberOf = memberOf
        if telephoneNumbers is not None:
            self.telephoneNumbers = telephoneNumbers
